test/test_env.cpp

```diff
@@ -0,0 +1,53 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <string>
+
+#include "rcutils/env.h"
+#include "rcutils/get_env.h"
+
+TEST(TestEnv, test_set_env) {
+  const char * res = nullptr;
+
+  // Invalid cases
+  EXPECT_FALSE(rcutils_set_env(nullptr, nullptr));
```

Comment:
You may want to call  `rcutils_reset_error();` when a error happens  otherwise you will show some overwritten errors in the terminal.

Reply:
Done in 8a6ee18